lib/logstash/outputs/scalyr.rb

```diff
@@ -331,8 +340,25 @@ def register
         @record_stats_for_status, @flush_quantile_estimates_on_status_send,
         @http_connect_timeout, @http_socket_timeout, @http_request_timeout, @http_pool_max, @http_pool_max_per_route
     )
-    send_status(initial_send_status_client_session)
-    initial_send_status_client_session.close
+
+    # We propagate errors on intial request to better handle errors related to invalid hostname
+    # or similar
+    begin
+      send_status(initial_send_status_client_session, true)
+    rescue Scalyr::Common::Client::ClientError => e
```

Comment:
We should probably also have a catch-all rescue here that logs the error as well, since even on startup we would want to ignore temporary server errors and other such things. Also I think thats the reason the unittests are failing.

Reply:
With "catch all rescue" you mean never actually propagating configuration error to the end user (and not failing plugin register) on error here?

In scalyr agent we do a similar thing in init / test connecting phase (before forking) we check the connection + API key and fail with fatal error in case the connection doesn't succeed (user can skip this step by passing ``--no-check-remote-server`` flag).

Having said that, I was also a bit undecided if we should throw and fail in this case or not. I do think it's probably fine since this happens as part of a plugin init / register phase (fail fast and early on config errors).

If we are worried about temporary server errors during this check, we could enable retry manticore functionality for this specific request.

Reply:
No I meant having a second `rescue` besides the ClientError one here, and doing the same `@logger.error` call in that. The issue is that if this initial check runs into a ServerError (or anything that isnt a ClientError) it bubbles all the way up. Crashing early on config errors makes sense but this will also crash on things besides that at the moment.

Reply:
So on that note, we have an option to just log an error (as you suggested) or perhaps as middle of the ground approach, having behavior as-is, but adding a config option to opt-out of it (similar to the ``--no-check-remote-server`` flag behavior).

Reply:
@yanscalyr ah, yeah, agreed - thanks makes sense.

Reply:
And while we are at it, it may also not be a bad idea to extend it and perform more complete "ping" check, something along the lines we do in the agent, and also verify the API key.

This way, plugin won't start accepting events (and potentially dropping them to the ground after exhausting all the retries) before everything is configured correctly.